API/cpp/Inc/I2C.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "PinDefs.hpp"
+
+#include <vector>
+
+typedef enum I2CBus {
+    I2CBus1,
+    I2CBus4,
+} I2CBus;
+
+class I2C {
+public:
+    I2C(I2CBus i2cBus);
+    ~I2C();
+
+    void transmit(uint8_t address, uint8_t regAddr, uint8_t data);
+    void transmit(uint8_t address, uint8_t regAddr, const std::vector<uint8_t>& data);
+
+    uint8_t receive(uint8_t address, uint8_t regAddr);
+    std::vector<uint8_t> receive(uint8_t address, uint8_t regAddr, size_t count);
```

Comment:
Why is there no recv-into-buffer API?

Reply:
(For context, my rant from robocup-firmware)

`<kyle dynamic allocation rant>`

I am not a fan of this. This heap-allocates a 2-byte value. Repeatedly. To read a register. AND the size is bounded at compile time (I assume) because you are passing in 2.

I would `strongly` prefer to at least have a second API for receive that allows you to pass in a pointer to a buffer and a size. Same with transmit above. This is overkill and physically pains me.

`</kyle dynamic allocation rant>`

Reply:
FWIW the SPI APIs are exactly what I want to see here.